src/core/basics.ml

```diff
@@ -100,13 +114,24 @@ let eq : term -> term -> bool = fun a b -> a == b ||
     | (Prod(a1,b1), Prod(a2,b2))
     | (Abst(a1,b1), Abst(a2,b2)) -> let (_, b1, b2) = Bindlib.unbind2 b1 b2 in
                                     eq ((a1,a2)::(b1,b2)::l)
+    | (LLet(t1,a1,u1), LLet(t2,a2,u2)) ->
+        let (_, u1, u2) = Bindlib.unbind2 u1 u2 in
+        eq ((a1,a2)::(t1,t2)::(u1,u2)::l)
     | (Appl(t1,u1), Appl(t2,u2)) -> eq ((t1,t2)::(u1,u2)::l)
     | (Meta(m1,e1), Meta(m2,e2)) when m1 == m2 ->
         eq (if e1 == e2 then l else List.add_array2 e1 e2 l)
     | (Wild       , _          )
     | (_          , Wild       ) -> eq l
     | (TRef(r)    , b          ) -> r := Some(b); eq l
     | (a          , TRef(r)    ) -> r := Some(a); eq l
+    (* Try unfolding variable definitions. *)
```

Comment:
Thanks to move the line `| (Vari(x1)   , Vari(x2)   ) when Bindlib.eq_vars x1 x2 -> eq l` here.